xapian-letor/ranker/ranker.cc

```diff
@@ -192,32 +193,31 @@ write_to_file(const std::vector<Xapian::FeatureVector> & list_fvecs, const strin
     }
 }
 
+// Query file is in the format: <qid> '<query_string>'
+// Although it will accept any number of arbitrary characters between
+// the first space and the first single quote.
 static std::pair<string, string>
 parse_query_string(const string & query_line, int line_number)
 {
-    vector<string> token;
-    size_t j = 0;
-    while (j < query_line.size()) {
-	size_t i = query_line.find_first_not_of(' ', j);
-	if (i == string::npos) break;
-	j = query_line.find_first_of(' ', i);
-	token.push_back(query_line.substr(i, j - i));
+    Assert(!query_line.empty());
+    string::size_type j = query_line.find_first_of(' ');
+    if (j == string::npos) {
```

Comment:
You should also check for `j == 0`. `          '123'` will be a valid query according to this.

Reply:
fixed in a fixup commit. I have also added a test case regarding the same.

Reply:
Message should be "Empty query id found " .... Test case should be just a space followed by query

>   'query'

Reply:
fixed in a fixup commit.